deps/rabbitmq_prometheus/src/collectors/prometheus_rabbitmq_global_metrics_collector.erl

```diff
@@ -0,0 +1,50 @@
+%% This Source Code Form is subject to the terms of the Mozilla Public
+%% License, v. 2.0. If a copy of the MPL was not distributed with this
+%% file, You can obtain one at https://mozilla.org/MPL/2.0/.
+%%
+%% Copyright (c) 2007-2021 VMware, Inc. or its affiliates.  All rights reserved.
+%%
+-module(prometheus_rabbitmq_global_metrics_collector).
+
+-behaviour(prometheus_collector).
+-include_lib("prometheus/include/prometheus.hrl").
+
+-export([register/0,
+         deregister_cleanup/1,
+         collect_mf/2]).
+
+-import(prometheus_model_helpers, [create_mf/4,
+                                   counter_metric/2]).
+
+%% This exposes the new global metrics which are instrumented directly, and bypass the Rabbit Core metrics entirely.
+%% The short-term plan is to start building the new metrics sub-system alongside RabbitMQ Core metrics
+%% The long-term plan is to replace RabbitMQ Core metrics with this new approach.
```

Comment:
Is my understanding correct that the new approach will only by used for aggregated counters, i.e. not for gauges and  not for per-object metrics?
Per-object metrics need to be handled by RabbitMQ Core metrics and can't be migrated to the new approach since the new approach "is best to only be used by global persistent counters that will never be deleted."?

Reply:
Yes, `rabbit_global_counters` is only meant to be used for aggregated counters. We want to track per-object metrics using seshat too, but via a different module (or maybe modules?).

The long-term plan is to extend the seshat library with per-object capabilities, and eventually deprecate the way we currently track metrics, including emitting them periodically, which creates a lot of overhead. @dcorbacho has been talking about this for years, and we finally started addressing it, which makes me super happy, and others too.

@deadtrickster will be joining this party soon 🙌🏻